### PR TITLE
Allow macros calling inherited methods on modules

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/Splicer.scala [hunk 1]
```diff
@@ -313,9 +313,14 @@ object Splicer {
...
+        } else if (fn.symbol.is(Module)) {
+          interpretModuleAccess(fn.symbol)
         } else if (fn.symbol.isStatic) {
-          if (fn.symbol.is(Module)) interpretModuleAccess(fn.symbol)
-          else interpretStaticMethodCall(fn.symbol, args.map(arg => interpretTree(arg)))
+          val module = fn.symbol.owner
+          interpretStaticMethodCall(module, fn.symbol, args.map(arg => interpretTree(arg)))
+        } else if (fn.qualifier.symbol.is(Module)) {
+          val module = fn.qualifier.symbol.moduleClass
+          interpretStaticMethodCall(module, fn.symbol, args.map(arg => interpretTree(arg)))
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/Splicer.scala]: How we can be sure that `module` is static?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/Splicer.scala]: I should add that condition

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/Splicer.scala]: Added

File: compiler/src/dotty/tools/dotc/transform/Splicer.scala [hunk 2]
```diff
@@ -313,9 +313,14 @@ object Splicer {
...
+        } else if (fn.symbol.is(Module)) {
+          interpretModuleAccess(fn.symbol)
         } else if (fn.symbol.isStatic) {
-          if (fn.symbol.is(Module)) interpretModuleAccess(fn.symbol)
-          else interpretStaticMethodCall(fn.symbol, args.map(arg => interpretTree(arg)))
+          val module = fn.symbol.owner
+          interpretStaticMethodCall(module, fn.symbol, args.map(arg => interpretTree(arg)))
+        } else if (fn.qualifier.symbol.is(Module) && fn.qualifier.symbol.isStatic) {
+          val module = fn.qualifier.symbol.moduleClass
+          interpretStaticMethodCall(module, fn.symbol, args.map(arg => interpretTree(arg)))
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/Splicer.scala]: Accessing `fn.qulifier` is dangerous (e.g., with import `Ident.qualifier.symbol` will return `NoSymbol`), it's more reliable to check types of `fun`, only handle the `TermRef` case. Then it seems the last two `else if` can be refactored and merged.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/Splicer.scala]: Here we never traverse imports, the only refTree are the ones in expressions.

Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/transform/Splicer.scala]: Sorry, I was not clear, I mean the case like the following:

```Scala
import o._
f(1)

// in constrast to
o.f(1)
```

The code seems to only handle the 2nd case, not the 1st one.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/Splicer.scala]: Oh, the second one will be generalized to the evaluation of any prefix in the future.

**Pull request merged**